test/Controller/ContentAlertsControllerTest.php

```diff
@@ -0,0 +1,154 @@
+        $this->assertSame(
+            ['Please provide your email address.', 'Please select an email type to subscribe.'],
+            array_map('trim', $crawler->filter('.form-item__message')->extract(['_text']))
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_requires_a_valid_email()
+    {
+        $client = static::createClient();
+
+        $crawler = $client->request('GET', '/content-alerts');
+
+        $form = $crawler->selectButton('Subscribe')->form();
+        $form['content_alerts[email]'] = 'foo';
+
+        $crawler = $client->submit($form);
+
+        $this->assertCount(1, $crawler->filter('.info-bar'));
+        $this->assertSame('There were problems submitting the form.', trim($crawler->filter('.info-bar')->text()));
+        $this->assertSame(
+            ['Please provide a valid email address.'],
+            array_map('trim', $crawler->filter('.form-item__message')->extract(['_text']))
+        );
+    }
+
+    /**
+     * @test
+     */
+    public function it_displays_confirmation_message()
+    {
+        $client = static::createClient();
+
+        $crawler = $client->request('GET', '/content-alerts');
+
+        $form = $crawler->selectButton('Subscribe')->form();
+        $form['content_alerts[email]'] = 'foo@bar.com';
+
+        $crawler = $client->submit($form);
+
+//        $this->assertEmpty($crawler->filter('.info-bar'));
+        $this->assertSame('Thank you for subscribing!', $crawler->filter('#thank-you h2')->text());
+        $this->assertSame('A confirmation email has been sent to foo@bar.com.', $crawler->filter('#thank-you p')->text());
+        $this->assertSame('Back to Homepage', $crawler->filter('#thank-you a')->text());
+        $this->assertSame('/', $crawler->filter('#thank-you a')->attr('href'));
+    }
+
+    /**
+     * @test
+     */
+    public function it_has_a_csrf_token_if_you_are_logged_in()
+    {
+        $client = static::createClient();
+
+        $this->logIn($client);
+
+        $crawler = $client->request('GET', '/content-alerts');
+
+        $form = $crawler->selectButton('Subscribe')->form();
+
+        $this->assertTrue($form->has('content_alerts[_token]'));
+    }
+
+    /**
+     * @test
+     */
+    public function it_has_a_honeypot()
+    {
+        $client = static::createClient();
+
+        $crawler = $client->request('GET', $this->getUrl());
+
+        $form = $crawler->selectButton('Subscribe')->form();
+        $form['content_alerts[email]'] = 'foo@example.com';
+
+        $form["content_alerts[{$this->getParameter('honeypot_field')}]"] = 'bar@example.com';
+        $crawler = $client->submit($form);
+
+        $this->assertCount(1, $crawler->filter('.info-bar'));
+        $this->assertSame('Please try submitting the form again.', trim($crawler->filter('.info-bar')->text()));
+    }
+
+    protected function getUrl() : string
+    {
+        return '/content-alerts';
+    }
+}
```

Comment:
the tests were very useful in describing the behavior, and you have got a Behat layer as well on top to cover the happy path